teerex/src/lib.rs

```diff
@@ -237,13 +245,37 @@ pub mod pallet {
 				sender_enclave.mr_enclave.encode() == shard_id.encode(),
 				<Error<T>>::WrongMrenclaveForShard
 			);
-			<WorkerForShard<T>>::insert(shard_id, sender_index);
-			log::debug!(
-				"Proposed sidechain block confirmed with shard {:?}, block hash {:?}",
-				shard_id,
-				block_hash
-			);
-			Self::deposit_event(Event::ProposedSidechainBlock(sender, block_hash));
+
+			let mut latest_block_header = <LatestSidechainBlockHeader<T>>::get();
+			let mut latest_block_number = latest_block_header.block_number;
+			let block_number = block_header.block_number;
+			if latest_block_number + 1 == block_number {
+				latest_block_header = Self::check_block_and_get_latest_header(
+					shard_id,
+					block_header,
+					&sender,
+					sender_index,
+					latest_block_header,
+				);
+				latest_block_number = latest_block_header.block_number;
+				while <SidechainBlockHeaderQueue<T>>::contains_key(latest_block_number + 1) {
```

Comment:
while loops in a blockchain dispatchable are very dangerous without a formal proof that this will not run for too long, this is not acceptable.

I want to incorporate here the `EARLY_BLOCK_PROPOSAL_LENIENCE` as an upper bound for iterations. (See comment below for that).